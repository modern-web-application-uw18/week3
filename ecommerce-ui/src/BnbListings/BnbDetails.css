@import './BnbListings.css';

.bnb-details {
  display: flex;
}

.bnb-details .media {
  max-width: 32rem;
}

.bnb-details .media img {
  width: 100%;
  border: .5px solid rgba(0,0,0,.1);
}

.bnb-details .content {
  padding: 1rem;
}

.bnb-details .bnb-stats {
  margin-bottom: 1.5rem;
}

@media screen and (max-width: 640px) {
  .bnb-details {
    flex-direction: column;
  }
}

@media screen and (min-width: 641px) {
  .bnb-details {
    flex-direction: row;
  }
}
