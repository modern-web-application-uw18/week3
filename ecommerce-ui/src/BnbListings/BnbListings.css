.bnb-listing {
  padding-bottom: .8rem;
  border-bottom: .5px solid rgb(68, 58, 45);
  margin-bottom: 2rem;
}

.bnb-listing .media {
  position: relative;
  background-position: center center;
  background-size: cover;
}

.bnb-listing .more-info {
  display: none;
}

.bnb-listing .media .more-info--text {
  color: white;
  font-size: .8rem
}

.bnb-listing .media .more-info--icon {
  height: 18px;
}

.bnb-listing .media .more-info:hover,
.bnb-listing .media:hover .more-info {
  display: block;
  position: absolute;
  margin-top: -2rem;
  background-color: rgb(68, 58, 45);
  bottom: 0;
  right: 0;
  width: auto;
  height: 2rem;
  padding: .3rem .6rem;
  display: flex;
  flex-direction: row;
  align-items: center;
}

.bnb-listing .content {
  padding-top: .8rem;
}

.bnb-details h1,
.bnb-listing h1 {
  font-family: 'Slabo 27px', serif;
  margin: .5rem 0 .5rem 0;
  color: rgb(68, 58, 45);
}

.bnb-listing h1 {
  font-size: 1.5rem;
}

.bnb-details h1 {
  font-size: 2rem;
  max-width: 20rem;
}

.bnb-details h2,
.bnb-listing h2 {
  margin-left: .1rem;
  font-size: .8rem;
  text-transform: uppercase;
  color: rgb(130, 110, 85);
}

.bnb-listing .rating-data > div,
.bnb-details p,
.bnb-listing p {
  font-size: .8rem;
  margin-left: .1rem;
}

.bnb-details .price,
.bnb-listing .price {
  font-weight: bold;
  color: rgb(68, 58, 45);
}

.bnb-details .bnb-stats,
.bnb-listing .bnb-stats {
  display: flex;
  flex-direction: row;
  align-items: center;
  margin-top: .5rem;
  justify-content: space-between;
}

.bnb-details .host-info,
.bnb-listing .host-info {
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 0 1rem;
}

.bnb-details .host-info a,
.bnb-listing .host-info a {
  text-decoration: none;
  color: rgb(130, 110, 85);
  font-weight: bold;
  font-size: .7rem;
}

.bnb-details .host-info a:hover,
.bnb-listing .host-info a:hover {
  text-decoration: underline;
}

.bnb-details .host-icon,
.bnb-listing .host-icon {
  margin-right: .3rem;
}

.bnb-details .host-info svg,
.bnb-listing .host-icon svg {
  color: rgb(130, 110, 85);
}

@media screen and (max-width: 640px) {
  .bnb-listing {
    width: 100%;
  }
  .bnb-listing .media {
    min-height: 10rem;
    height: auto;
  }
}

@media screen and (min-width: 641px) {
  .bnb-listing {
    width: 49%;
  }
  .bnb-listing .media {
    height: 11rem;
  }
}

@media screen and (min-width: 768px) {
  .bnb-listing {
    width: 32%;
  }
}

/*@media screen and (min-width: 1024px {
  .bnb-listing {
    width: 32%;
  }
}*/
