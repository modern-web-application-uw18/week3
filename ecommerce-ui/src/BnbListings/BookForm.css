.book-form {
  width: 100%;
  border: .5px solid rgba(0,0,0,.4);
  padding: .8rem;
}

.book-form label > span {
  display: block;
  color: rgb(130, 110, 85);
  margin-bottom: .5rem;
}

.book-form label input {
  margin-bottom: .8rem;
}

.book-form input[type="submit"] {
  background-color: rgb(130, 110, 85);
  color: white;
  border: none;
  display: block;
  font-size: 1rem;
  padding: .3rem .5rem;
  font-family: 'Slabo 27px', serif;
}
