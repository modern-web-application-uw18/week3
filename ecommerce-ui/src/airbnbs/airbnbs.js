import React, { Component } from 'react';
import airbnbs from './airbnbs.json';
import AddRental from '../addRental/addRental';
import Cart from '../cart/cart';
import '../App.css';
import CartTotal from '../cartTotal/cartTotal';

class Airbnblocation extends Component{
    constructor(props){
        super(props);
        this.state= {
            locations: airbnbs,
            cart:[],
        };
    }


  addRental = (data) => {
      this.setState((prevState, props) => {
          const cart = data;
          const newCart = prevState.cart;
          newCart.push(cart);
          return {
              cart: newCart,
          };
      });
  }

   deleteRental = (index) => {
    this.setState((prevState, props) => {
        const newCart = prevState.cart;
        newCart.splice(index, 1);
        return {
            cart: newCart
        };
    });
}

      render() {
        let locations = this.state.locations.map((locations, idx) => {
          return(
              <AddRental key={idx}
              addRental={this.addRental}
              image={locations.image}
              title={locations.title}
              stars={locations.rating.stars}
              numberofreviews={locations.rating.reviews}
              houseType={locations.houseType}
              city={locations.location.city}
              country={locations.location.country}
              cost={locations.payment.cost}
              description={locations.payment.description}
              host={locations.host.name}
              issuperhost={locations.host.isSuperhost}
              />
            )
      })
      let shoppingCart = this.state.cart.map((cart, idx) => {   
          return(
            <Cart key={idx}
            index={idx}
            deleteRental={this.deleteRental}
            image={cart.image}
            title={cart.title}
            stars={cart.stars}
            numberofreviews={cart.reviews}
            houseType={cart.houseType}
            city={cart.city}
            country={cart.country}
            cost={cart.cost}
            description={cart.description}
            host={cart.name}
            issuperhost={cart.isSuperhost}
            />
          )
    })
      return (

        <div>
          <h3>Cart</h3>
          <CartTotal
          cost={this.state.cart}/>
          <div  className="flex-container">
          {shoppingCart}
          </div>
          <h3>Pick your destination!</h3>
          <div className="flex-container">{locations}</div>
        </div>

      )
    }
    }

    export default Airbnblocation;
