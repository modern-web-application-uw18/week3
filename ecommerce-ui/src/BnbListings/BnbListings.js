import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Link,
  BrowserRouter
} from "react-router-dom";
import './BnbListings.css';
import SuperHostIcon from '../HostIcon/SuperHostIcon';
import HostIcon from '../HostIcon/HostIcon';

class BnbListings extends Component {

  getHostIcon() {
    return (
      <div className="host-icon">
        <HostIcon />
      </div>
    );
  };

  getSuperHostIcon() {
    return (
      <div className="host-icon">
        <SuperHostIcon />
      </div>
    );
  };

  getPaymentDescription(descript) {
    return (
      <span>
        · {descript}
      </span>
    );
  };

  render() {
    const isSuperHost = this.props.bnb.host.isSuperhost;
    const image = this.props.bnb.image;
    const backgroundMediaStyles = {
        backgroundImage: 'url(' + image + ')',
    };
    const starNumber = this.props.bnb.rating.stars;
    const paymentDescription = this.props.bnb.payment.description;
    return (
      <BrowserRouter>
      <section className="bnb-listing">
        <Link to={`/id/${this.props.bnb.id}`} style={{ textDecoration: 'none' }}>
          <div className="media" style={backgroundMediaStyles}>
            <div className="more-info">
              <span className="more-info--text">Learn More</span>
              <span className="more-info--icon">
                <svg xmlns="http://www.w3.org/2000/svg" width="18" height="18" viewBox="0 0 24 24" fill="none" stroke="white" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round">
                  <line x1="12" y1="5" x2="12" y2="19">
                  </line>
                  <line x1="5" y1="12" x2="19" y2="12">
                  </line>
                </svg>
              </span>
            </div>
          </div>
        </Link>
        <div className="content">
          <Link to={`/id/${this.props.bnb.id}`} style={{ textDecoration: 'none' }}>
            <h2>{this.props.bnb.houseType} · {this.props.bnb.location.city}, {this.props.bnb.location.country}</h2>
            <h1>{this.props.bnb.title}</h1>
            <p><span className="price">${this.props.bnb.payment.cost}</span> per night {paymentDescription ? this.getPaymentDescription(paymentDescription) : null}</p>
          </Link>
          <div className="bnb-stats">
            <div className="rating-data">
              <div>Rating: {starNumber} stars / {this.props.bnb.rating.reviews} reviews</div>
            </div>
            <div className="host-info" alt={this.props.bnb.host.name} title={this.props.bnb.host.name}>
              {isSuperHost ? this.getSuperHostIcon() : this.getHostIcon() }
              <a href="/">{this.props.bnb.host.name}</a>
            </div>
          </div>
        </div>
      </section>
      </BrowserRouter>
    );
  }
}

export default BnbListings;

BnbListings.propTypes = {
  bnb: PropTypes.shape({
    title: PropTypes.string.isRequired,
    houseType: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    location: PropTypes.object.isRequired,
    location: PropTypes.shape({
      city: PropTypes.string,
      country: PropTypes.string
    }),
    payment: PropTypes.object,
    payment: PropTypes.shape({
      cost: PropTypes.number,
      description: PropTypes.string
    }),
    host: PropTypes.object,
    host: PropTypes.shape({
      name: PropTypes.string,
      isSuperhost: PropTypes.bool
    }),
    rating: PropTypes.object,
    rating: PropTypes.shape({
      stars: PropTypes.number,
      reviews: PropTypes.number
    })
  })
};
