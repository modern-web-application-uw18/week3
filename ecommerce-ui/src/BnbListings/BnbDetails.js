import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './BnbDetails.css';
import airbnbs from '../airbnbs.json';
import SuperHostIcon from './SuperHostIcon';
import HostIcon from './HostIcon';
import BookForm from './BookForm';

class BnbDetails extends Component {
  constructor(props) {
    super(props);
  }

  getHostIcon() {
    return (
      <div className="host-icon">
        <HostIcon />
      </div>
    );
  }

  getSuperHostIcon() {
    return (
      <div className="host-icon">
        <SuperHostIcon />
      </div>
    );
  }

  getPaymentDescription(descript) {
    return (
      <span>
        · {descript}
      </span>
    );
  };

  render() {
    const itemID = this.props.match.params.id;
    const obj = airbnbs.find(function (obj) { return obj.id === itemID;});
    const isSuperHost = obj.host.isSuperhost;
    const paymentDescription = obj.payment.description;
    return (
      <section className="bnb-details">
        <div className="media">
          <img src={obj.image} alt={obj.title} title={obj.title}/>
        </div>
        <div className="content">
        <h2>{obj.houseType} · {obj.location.city}, {obj.location.country}</h2>
        <h1>{obj.title}</h1>
        <p><span className="price">${obj.payment.cost}</span> per night  {paymentDescription ? this.getPaymentDescription(paymentDescription) : null}</p>
        <div className="bnb-stats">
          <div className="rating-data">
            <p>Rating: {obj.rating.stars} stars / {obj.rating.reviews} reviews</p>
          </div>
          <div className="host-info" alt={obj.host.name} title={obj.host.name}>
            {isSuperHost ? this.getSuperHostIcon() : this.getHostIcon() }
            <a href="/">{obj.host.name}</a>
          </div>
        </div>
        <BookForm bnb={obj} addItemToCart={this.props.addItemToCart}/>
        </div>
      </section>

    );
  }
}

export default BnbDetails;

BnbDetails.propTypes = {
  addItemToCart: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  houseType: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  location: PropTypes.object.isRequired,
  location: PropTypes.shape({
    city: PropTypes.string,
    country: PropTypes.string
  }),
  payment: PropTypes.object,
  payment: PropTypes.shape({
    cost: PropTypes.number,
    description: PropTypes.string
  }),
  host: PropTypes.object,
  host: PropTypes.shape({
    name: PropTypes.string,
    isSuperhost: PropTypes.bool
  }),
  rating: PropTypes.object,
  rating: PropTypes.shape({
    stars: PropTypes.number,
    reviews: PropTypes.number
  })
};
