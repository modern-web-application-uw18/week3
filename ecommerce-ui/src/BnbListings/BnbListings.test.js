import React from 'react';
import BnbListings from './BnbListings';
import ReactTestRenderer from 'react-test-renderer';
import airbnbs from '../airbnbs.json';
import {
  MemoryRouter
} from "react-router-dom";

describe('BnbListings component', () => {
  it('should render', () => {
    const component = ReactTestRenderer.create(
      <MemoryRouter>
        <BnbListings bnb={airbnbs[0]} />
      </MemoryRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
