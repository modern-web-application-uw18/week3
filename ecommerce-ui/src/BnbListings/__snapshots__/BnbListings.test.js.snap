// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BnbListings component should render 1`] = `
<section
  className="bnb-listing"
>
  <a
    href="/id/1490806230066"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <div
      className="media"
      style={
        Object {
          "backgroundImage": "url(https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80)",
        }
      }
    >
      <div
        className="more-info"
      >
        <span
          className="more-info--text"
        >
          Learn More
        </span>
        <span
          className="more-info--icon"
        >
          <svg
            fill="none"
            height="18"
            stroke="white"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="5"
              y2="19"
            />
            <line
              x1="5"
              x2="19"
              y1="12"
              y2="12"
            />
          </svg>
        </span>
      </div>
    </div>
  </a>
  <div
    className="content"
  >
    <a
      href="/id/1490806230066"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <h2>
        Apartment
         · 
        New York
        , 
        USA
      </h2>
      <h1>
        Centrally-located Manhattan studio
      </h1>
      <p>
        <span
          className="price"
        >
          $
          105
        </span>
         per night 
        <span>
          · 
          Free cancellation
        </span>
      </p>
    </a>
    <div
      className="bnb-stats"
    >
      <div
        className="rating-data"
      >
        <div>
          Rating: 
          3
           stars / 
          133
           reviews
        </div>
      </div>
      <div
        alt="Robert"
        className="host-info"
        title="Robert"
      >
        <div
          className="host-icon"
        >
          <svg
            fill="none"
            height="24"
            stroke="rgb(130, 110, 85)"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 21v-2a4 4 0 0 0-4-4H5a4 4 0 0 0-4 4v2"
            />
            <circle
              cx="8.5"
              cy="7"
              r="4"
            />
            <line
              x1="20"
              x2="20"
              y1="8"
              y2="14"
            />
            <line
              x1="23"
              x2="17"
              y1="11"
              y2="11"
            />
          </svg>
        </div>
        <a
          href="/"
        >
          Robert
        </a>
      </div>
    </div>
  </div>
</section>
`;
